composeml/label_times/object.py

```diff
@@ -12,70 +12,102 @@
 
-    def _check_label_name(self):
-        """Checks whether the target exists in the data frame."""
-        if self.label_name is None:
-            self.label_name = self._infer_label_name()
+        if self.target_types.empty:
+            self.target_types = self._infer_target_types()
+        else:
+            target_names = self.target_types.index.tolist()
+            match = target_names == self.target_columns
+            assert match, 'target names in types must match target columns'
 
-        info = 'target variable not found: %s' % self.label_name
-        assert self.label_name in self.columns, info
+    def _check_label_times(self):
+        """Validates the lables times object."""
+        self._check_target_columns()
+        self._check_target_types()
 
-    def _infer_label_name(self):
-        """Infers the target name from the data frame.
+    def _infer_target_columns(self):
+        """Infers the names of the targets in the data frame.
 
         Returns:
-            value (str): Inferred target name.
+            value (list): A list of the target names.
         """
         not_targets = [self.target_entity, 'time']
-        target_names = self.columns.difference(not_targets)
-        value = target_names.tolist()[0]
+        target_columns = self.columns.difference(not_targets)
+        assert not target_columns.empty, 'target columns not found'
+        value = target_columns.tolist()
         return value
 
-    def _check_label_type(self):
-        """Checks whether the target type is continuous or discrete."""
-        if self.label_type is None:
-            self.label_type = self._infer_label_type()
-
-        error = 'label type must be "continuous" or "discrete"'
-        assert self.label_type in ['continuous', 'discrete'], error
-
-    def _infer_label_type(self):
-        """Infers the target type from the data type.
+    @property
+    def _is_single_target(self):
+        return len(self.target_columns) == 1
 
-        Returns:
-            value (str): Inferred label type. Either "continuous" or "discrete".
-        """
-        dtype = self[self.label_name].dtype
+    def _get_target_type(self, dtype):
         is_discrete = pd.api.types.is_bool_dtype(dtype)
         is_discrete |= pd.api.types.is_categorical_dtype(dtype)
         is_discrete |= pd.api.types.is_object_dtype(dtype)
         value = 'discrete' if is_discrete else 'continuous'
         return value
 
+    def _infer_target_types(self):
+        """Infers the target type from the data type.
+
+        Returns:
+            types (Series): Inferred label type. Either "continuous" or "discrete".
+        """
+        dtypes = self.dtypes[self.target_columns]
+        types = dtypes.apply(self._get_target_type)
+        return types
+
+    def select(self, target):
+        assert not self._is_single_target, 'only one target exists'
+        if not isinstance(target, str): raise TypeError('target name must be string')
```

Comment:
This change looks good, but we should probably also add a test case to verify this Error gets raised properly.

Reply:
Test case was added (2a7904e).